packages/blade/src/components/Modal/Modal.web.tsx

```diff
@@ -0,0 +1,245 @@
+      );
+    }
+  }, [platform]);
+
+  // required by floating ui to handle focus
+  const { refs, context } = useFloating({
+    open: isMounted,
+  });
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const defaultInitialFocusRef = React.useRef<any>(null);
+  const originalFocusElement = React.useRef<HTMLElement | null>(null);
+
+  const returnFocus = React.useCallback(() => {
+    if (!originalFocusElement.current) return;
+    originalFocusElement.current.focus();
+    // After returning focus we will clear the original focus
+    // Because if modal can be opened up via multiple triggers
+    // We want to ensure the focus returns back to the most recent triggerer
+    originalFocusElement.current = null;
+  }, [originalFocusElement]);
+
+  const focusOnInitialRef = React.useCallback(() => {
+    if (!initialFocusRef) {
+      // focus on close button
+      defaultInitialFocusRef.current?.focus();
+    } else {
+      // focus on the initialRef passed by the user
+      initialFocusRef.current?.focus();
+    }
+  }, [initialFocusRef]);
+
+  React.useEffect(() => {
+    if (isOpen) {
+      // set the original focus element where the focus will return to after closing the modal
+      originalFocusElement.current =
+        originalFocusElement.current ?? (document.activeElement as HTMLElement);
+      // focus on an element on Modal, if initialFocusRef is not passed, focus on the close button
+      focusOnInitialRef();
+    }
+  }, [isOpen, focusOnInitialRef]);
+
+  React.useEffect(() => {
+    // Return focus to the element that originally had it
+    if (!isOpen) {
+      returnFocus();
+    }
+  }, [isOpen, returnFocus]);
+
+  const modalContext = React.useMemo(
+    () => ({
+      isOpen,
+      close: onDismiss,
+      defaultInitialFocusRef,
+      isVisible,
+      footerHeight,
+      setFooterHeight,
+    }),
+    [isOpen, onDismiss, defaultInitialFocusRef, isVisible, footerHeight, setFooterHeight],
+  );
+  const handleKeyDown = (event: React.KeyboardEvent): void => {
+    // close modal on escape key press
+    if (event?.key === 'Escape' || event?.code === 'Escape') {
+      onDismiss();
+    }
+  };
+
+  // Only allow ModalHeader, ModalBody and ModalFooter as children
+  const validChildren = React.Children.map(children, (child) => {
+    if (
+      isValidAllowedChildren(child, MetaConstants.ModalHeader) ||
+      isValidAllowedChildren(child, MetaConstants.ModalBody) ||
+      isValidAllowedChildren(child, MetaConstants.ModalFooter)
+    ) {
+      return child;
+    } else {
+      throw new Error(
+        '[Blade Modal] Modal only accepts ModalHeader, ModalBody and ModalFooter as children',
+      );
+    }
+  });
+
+  return (
+    <ModalPortal>
+      <ModalContext.Provider value={modalContext}>
+        {isMounted ? (
+          <FloatingFocusManager context={context} modal={true}>
+            <Box zIndex={999} position="fixed">
```

Comment:
Can we add a TODO here for future reference? 

IMO as with all the things we should keep zIndexes in check, We should also expose zIndex (elevation rip naming) tokens and have proper guidelines on stacking context on which components will take zIndex priority. 

eg: 

Popovers: 2000
Tooltip: 1000
BottomSheet: 500
Modal: 500


In most of our products people are using random zindexes and because of that there were few zindex issues anshul faced while implementing BottomSheet, where from their end they were using a ridiculously high zindex values.

Reply:
As a starting point I'm abstracting Modal's zIndex into a common place. We should tokenize this